<launch>
    <arg name="output"             default="screen"    doc="display output to screen or log file"/>
    <arg name="odom_topic"         default="/slam_pose"    doc="ROS topic for the pose"/>
    <arg name="control_topic"      default="/control/servo_control"    doc="ROS topic to servo control"/>
    <arg name="obstacle_topic"     default="/prediction/obstacles"  doc="ROS topic for obstacles"/>
    <arg name="traj_topic"         default="/planning/trajectory"  doc="ROS topic for trajectory"/>
    <arg name="path_topic"         default="/planning/path"        doc="ROS topic for reference path"/>
    <arg name="predefined_path"    default="true"        doc="if true, load a predefined path from file rather than subscribing to a topic"/>

    <rosparam command="load" file="$(find traj_planning_ros)/configs/config.yaml" />

    <node pkg="traj_planning_ros" type="traj_planning_node.py" name="traj_planning" output="$(arg output)">
        <param name="odom_topic"           value="$(arg odom_topic)" />
        <param name="control_topic"        value="$(arg control_topic)" />
        <param name="obstacle_topic"       value="$(arg obstacle_topic)" />
        <param name="traj_topic"           value="$(arg traj_topic)" />
        <param name="path_topic"           value="$(arg path_topic)" />
        <param name="predefined_path"      value="$(arg predefined_path)" />
        <param name="package_path"               value="$(find traj_planning_ros)" />
    </node>

</launch>
